src/creator.rs

```diff
@@ -138,9 +139,13 @@ impl<H: Hasher> Creator<H> {
                     break;
                 },
             };
-            if delay_passed && self.check_ready() {
-                self.create_unit();
-                delay_fut = Delay::new((self.create_lag)(round + 1)).fuse();
+            while delay_passed && round < self.current_round {
```

Comment:
I think `delay_passed` is not necessary here – we always want to be no more than one round behind.